torch/distributed/distributed_c10d.py

```diff
@@ -1859,10 +1859,9 @@ def _new_process_group_helper(
             "created, please use a different group name"
         )
 
-    if device_id is not None and (device_id.index is None or device_id.type != "cuda"):
+    if device_id is not None and device_id.index is None:
```

Comment:
The new change does not gate all cases as the original one. What's the behavior of `device_id.type == "xpu"`?

Reply:
If a `device_id` is specified but `device_id.index` is none, then an exception will be raised. This logic also applies to `XPU` device.

Reply:
I think @EikanWang is right. The new code doesn't cover `torch.device('cpu', index=1)`.

Reply:
Got you. `torch.device('cpu', index=1)` is a very special CPU case, which should be raised with exception when a `device_id` is passed.